vpr/src/device/rr_graph_view.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RR_GRAPH_VIEW_H
+#define RR_GRAPH_VIEW_H
+
+#include "rr_graph_storage.h"
+#include "rr_spatial_lookup.h"
+
+/* An read-only routing resource graph
+ * which is an unified object including pointors to
+ * - node storage
+ * - TODO: edge_storage
+ * - TODO: node_ptc_storage
+ * - TODO: node_fan_in_storage
+ * - rr_node_indices
+ *
+ * Note that the RRGraphView does not own the storage
+ * It serves a virtual read-only protocol for
+ * - placer
+ * - router
+ * - timing analyzer
+ * - GUI
+ *
+ * Note that each client of rr_graph may get a frame view of the object
+ * The RRGraphView is the complete frame view of the routing resource graph
+ * - This helps to reduce the memory footprint for each client
+ * - This avoids massive changes for each client on using the APIs
+ *   as each frame view provides adhoc APIs for each client
+ *
+ * TODO: more compact frame views will be created, e.g., 
+ * - a mini frame view: contains only node and edges, representing the connectivity of the graph
+ * - a geometry frame view: an extended mini frame view with node-level attributes, 
+ *                          in particular geometry information (type, x, y etc).
+ *
+ */
+class RRGraphView {
+    /****************
+     * Constructors
+     ****************/
+  public:
+    RRGraphView(const t_rr_graph_storage& node_storage,
+                const RRSpatialLookup& node_lookup);
+
+    /* Disable copy constructors
+     * This is to avoid any duplication of the object
+     * as it is only interface allowed to access routing resource graph
+     */
+    RRGraphView(const RRGraphView&) = delete;
+
+    /* Disable copy assignment operator */
+    void operator=(const RRGraphView&) = delete;
+
+    /****************
+     * Accessors
+     ****************/
+  public:
+    /* Get the type of a routing resource node */
```

Comment:
Probably good to add a TODO comment saying that the goal is to have these accessors private at some point totally replaced with the 'questioin' kind of methods.

Reply:
Yes. The TODO comment is useful. But not every API will be converted to the ``questionin`` methods. But the ``questionin`` style is preferred.